feat(portal): Internet Sites

AndrewDryga opened the pull request:
Related #6834

AndrewDryga commented on elixir/apps/domain/lib/domain/auth/adapters/okta/api_client.ex:
```diff
@@ -100,13 +100,19 @@ defmodule Domain.Auth.Adapters.Okta.APIClient do
     end
   end
 
+  if Mix.env() == :test do
```
@bmanifold this trick allows to override the behaviour per environment. We should not put static sleeps anywhere - instead they should be either in a function that is overridden (like here, for temporary code that will be removed) or added to application environment (that we can override in `test.exs`).

bmanifold replied:
👍 makes sense.  When I added the sleep I was only thinking about making sure we weren't causing any rate limits to be triggered, but this is definitely a good thing for me to keep in mind.

bmanifold replied:
Out of curiosity, was this causing issues in the tests?  Or just slowing the tests down?

AndrewDryga replied:
Yeah, some cases were taking 10+ seconds by themselves while all suite before took about the same time.

bmanifold replied:
Ah, yeah, that's not good.  Sorry about that.  I think I must have let CI run the tests for me when I pushed that change, so I didn't feel the immediate pain of the tests taking longer.

AndrewDryga replied:
No worries, I just brought it up for the future

AndrewDryga commented on elixir/apps/web/lib/web/live/sites/gateways/index.ex:
```diff
@@ -65,7 +65,11 @@ defmodule Web.Sites.Gateways.Index do
       </:title>
-      <:help>
+      <:help :if={@group.managed_by == :system and @group.name == "Internet"}>
+        Gateways deployed to the Internet Site are used to route Internet Resource,
```
@jamilbk better content is welcome

jamilbk reviewed (commented):
I think this direction is the right one. How are existing Gateways treated with this change? How should we roll this out to customers? Are the existing policies for the old Internet Resource kept in tact?

Just wondering what the best way to roll this out is going to be.

We'll be asking customers to increase the number of VMs ($$) they run for Firezone, so I'll keep that in mind too.

jamilbk commented on elixir/apps/web/lib/web/live/sites/index.ex:
```diff
@@ -165,14 +179,67 @@ defmodule Web.Sites.Index do
+      </:action>
+
+      <:help>
+        This is a specialized Site used to host Gateways that tunnel traffic that does not match any Resource.
```
```suggestion
        This is a dedicated Site for traffic that does not match any specific Resource. Deploy Gateways here to enable full-route tunneling.

```

jamilbk commented on elixir/apps/web/lib/web/live/sites/show.ex:
```diff
@@ -97,12 +184,17 @@ defmodule Web.Sites.Show do
       </:action>
 
+      <:help :if={@group.managed_by == :system and @group.name == "Internet"}>
+        The Internet Site is a specialized Site used to host gateways that tunnel traffic that doesn't match any specific Resource.
```
```suggestion
        The Internet Site is a dedicated Site for traffic that does not match any specific Resource. 
```

jamilbk commented on elixir/apps/web/lib/web/live/sites/show.ex:
```diff
@@ -184,7 +279,15 @@ defmodule Web.Sites.Show do
+                      class={[link_style()]}
+                      navigate={~p"/#{@account}/sites/#{@group}/new_token"}
+                    >
+                      Deploy a gateway to the Internet site.
```
```suggestion
                      Deploy a Gateway to the Internet Site.
```

jamilbk commented on elixir/apps/web/lib/web/live/sites/index.ex:
```diff
@@ -165,14 +179,67 @@ defmodule Web.Sites.Index do
+
+      <:action :if={Domain.Accounts.internet_resource_enabled?(@account)}>
+        <.edit_button navigate={~p"/#{@account}/sites/#{internet_gateway_group}"}>
+          Manage Internet Tunneling
```
```suggestion
          Manage Full-route Tunneling
```

jamilbk reviewed (approved):
LGTM! Just have some minor text clarification suggestions. I think most admins have used the term `full-route` to refer to this so just added some suggestions to make that association more clear.

jamilbk committed:
Add Internet Sites

commit 73c07815b5bdbedb205721743daaf1534ee9e7b4
Merge: 61fa98425 df8b615d3
Author: Jamil Bou Kheir <jamilbk@users.noreply.github.com>
Date:   Fri Feb 14 16:15:47 2025 -0800

    Merge branch 'main' into andrew/internet-sites

commit 61fa9842539e630f78e94893b492e634ec7ed351
Author: Jamil <jamilbk@users.noreply.github.com>
Date:   Fri Feb 14 16:11:23 2025 -0800

    refactor(portal): Allow user to migrate internet resource (#8133)

    Building on top of #6905, we add a `Migrate Internet Resource` button
    that allows the admin to preemptively migrate their Internet Resource to
    the new Internet Site.

    <img width="1233" alt="Screenshot 2025-02-13 at 11 10 19 PM"
    src="https://github.com/user-attachments/assets/25036e9e-af9a-4515-952e-344b25a53d63"
    />

    <img width="728" alt="Screenshot 2025-02-13 at 11 09 50 PM"
    src="https://github.com/user-attachments/assets/13ea42af-5fb1-4e9d-b96c-9f28c6a91f9d"
    />

    ---------

    Signed-off-by: Jamil <jamilbk@users.noreply.github.com>
    Co-authored-by: Brian Manifold <bmanifold@users.noreply.github.com>

commit 1948faa19f004d887eb309f4301c84ab8b2f808d
Merge: e8dbafb57 0e5d91e26
Author: Jamil Bou Kheir <jamilbk@users.noreply.github.com>
Date:   Wed Feb 12 18:44:33 2025 -0800

    Merge branch 'main' into andrew/internet-sites

commit e8dbafb57f5a8c554aea3f7e3dac7977bfcbe6ca
Author: Andrew <a@firezone.dev>
Date:   Thu Oct 3 10:31:48 2024 -0600

    Allow created_by: system for the resource connections

commit 8bd849f2077a279e979f6399950c093f87ad450f
Author: Andrew <a@firezone.dev>
Date:   Thu Oct 3 10:03:57 2024 -0600

    Show and allow managing internet resources for some time

commit f033eb13a11440bb4e21c7e5c98bc9d32db2918b
Author: Andrew <a@firezone.dev>
Date:   Wed Oct 2 12:26:51 2024 -0600

    Update internet site status in realtime too

commit 225436ed39c4776ede05918100edadea7043937c
Author: Andrew <a@firezone.dev>
Date:   Wed Oct 2 11:52:27 2024 -0600

    Tests

commit 3b615c559849ecd5725eff2c4d2b40379c08e8fc
Author: Andrew <a@firezone.dev>
Date:   Tue Oct 1 16:12:38 2024 -0600

    UI to manage internet sites

commit 705bb08cf512eca18d2712d8dfb5239635975ed2
Author: Andrew <a@firezone.dev>
Date:   Tue Oct 1 12:27:05 2024 -0600

    Fix some slow tests

commit b2916fd817f2ad29e147f92c75f177f63a7f1667
Author: Andrew <a@firezone.dev>
Date:   Tue Oct 1 12:17:41 2024 -0600

    Create Internet site on Sign Up
